crates/utils/src/rate_limit/mod.rs

```diff
@@ -85,7 +84,7 @@ impl RateLimited {
       RateLimitType::Comment => (rate_limit.comment, rate_limit.comment_per_second),
       RateLimitType::Search => (rate_limit.search, rate_limit.search_per_second),
     };
-    let mut limiter = self.rate_limiter.lock();
+    let mut limiter = self.rate_limiter.lock().expect("mutex poison error");
```

Comment:
I was also a bit sceptical about this, but i think its fine. [Documentation](https://doc.rust-lang.org/std/sync/struct.Mutex.html#method.lock) says it will return an error "If another user of this mutex panicked while holding the mutex". I dont think thats a possibility here, as the mutex is only held for check_rate_limit_full, which does nothing but arithmetics.